Source/CarthageKit/Xcode.swift

```diff
@@ -885,27 +885,36 @@ public func buildScheme(scheme: String, withConfiguration configuration: String,
 			let values = map(sdksByPlatform) { ($0, $1) }
 			return SignalProducer(values: values)
 		}
-		|> flatMap(.Concat) { platform, sdks in
+		|> flatMap(.Concat) { platform, sdks -> SignalProducer<(Platform, [SDK], Bool), CarthageError> in
+			let filterResult = sdkFilter(sdks: sdks, scheme: scheme, configuration: configuration, project: project)
+
+			if let filteredSDKs = filterResult.value where filteredSDKs.isEmpty {
```

Comment:
This is required to let platform filter work correctly. Without this, tvOS platform contained in a universal framework target with `--platform ios` would crash with reaching to [this line](https://github.com/Carthage/Carthage/pull/798/files#diff-f66644e8cfbcad3c3826395a690ddfbcR907), for example.